src/ALabel.cpp

```diff
@@ -14,30 +15,129 @@ waybar::ALabel::ALabel(const Json::Value& config, const std::string format)
   }
   if (config_["format-alt"].isString()) {
     event_box_.add_events(Gdk::BUTTON_PRESS_MASK);
-    event_box_.signal_button_press_event()
-      .connect(sigc::mem_fun(*this, &ALabel::handleToggle));
+    event_box_.signal_button_press_event().connect(
+        sigc::mem_fun(*this, &ALabel::handleToggle));
+  }
+
+  // configure events' user commands
+  if (config_["on-click"].isString()) {
+    std::string cmd = config_["on-click"].asString();
+    setEventCmd(waybar::EventType::BUTTON_PRESS, cmd);
+  }
+  if (config_["on-scroll-up"].isString()) {
+    std::string cmd = config_["on-scroll-up"].asString();
+    setEventCmd(waybar::EventType::SCROLL_UP, cmd);
+  }
+  if (config_["on-scroll-down"].isString()) {
+    std::string cmd = config_["on-scroll-down"].asString();
+    setEventCmd(waybar::EventType::SCROLL_DOWN, cmd);
   }
 }
 
-auto waybar::ALabel::update() -> void
-{
+auto waybar::ALabel::update() -> void {
   // Nothing here
 }
 
-bool waybar::ALabel::handleToggle(GdkEventButton* const& /*ev*/)
-{
-  alt = !alt;
-  if (alt) {
-    format_ = config_["format-alt"].asString();
-  } else {
-    format_ = default_format_;
+bool waybar::ALabel::handleToggle(GdkEventButton* const& e) {
+  bool ret = false;
+  if (e->button == 1) {
+    ret = forkExec(button_press_cmd_);
+  } else if (e->button == 2) {
+    alt = !alt;
+    if (alt) {
+      format_ = config_["format-alt"].asString();
+    } else {
+      format_ = default_format_;
+    }
+    ret = true;
   }
+
   dp.emit();
+  return ret;
+}
+
+bool waybar::ALabel::forkExec(std::string cmd) {
```

Comment:
Moves this function in the util header file, it will always be useful.

Reply:
Sure, will do that.